power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -1440,8 +1265,11 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
 
             // Check the branch and shunt indices
             constexpr auto branch_param_in_seq_map =
-                std::array{index_of_component<Line>, index_of_component<Link>, index_of_component<Transformer>};
-            constexpr auto shunt_param_in_seq_map = std::array{index_of_component<Shunt>};
+                std::array{main_core::utils::index_of_component<Line, ComponentType...>,
+                           main_core::utils::index_of_component<Link, ComponentType...>,
+                           main_core::utils::index_of_component<Transformer, ComponentType...>};
```

Comment:
oof :joy: (maybe in the follow-up PR) can you make it a shorthand within this function? (may not be possible because of template stuff) or something along the lines of

```cpp
constexpr auto index_of = []<typename CompType> { return index_of_component<CompType, ComponentType...>; };
```

Reply:
Follow-up PR.